src/Middleware/Diagnostics/src/ExceptionHandler/ExceptionHandlerMiddleware.cs

```diff
@@ -113,11 +113,21 @@ private async Task HandleException(HttpContext context, ExceptionDispatchInfo ed
                 };
                 context.Features.Set<IExceptionHandlerFeature>(exceptionHandlerFeature);
                 context.Features.Set<IExceptionHandlerPathFeature>(exceptionHandlerFeature);
-                context.Response.StatusCode = 500;
+                context.Response.StatusCode = StatusCodes.Status500InternalServerError;
                 context.Response.OnStarting(_clearCacheHeadersDelegate, context.Response);
 
                 await _options.ExceptionHandler(context);
 
+                if (context.Response.StatusCode == StatusCodes.Status404NotFound)
+                {
+                    // Reset status and clear headers
+                    context.Response.StatusCode = StatusCodes.Status500InternalServerError;
+                    await ClearCacheHeaders(context.Response);
+
+                    // This exception will be logged and the original exception wil be thrown
+                    throw new InvalidOperationException($"No exception handler was found at the configured path {_options.ExceptionHandlingPath}.");
```

Comment:
ExceptionHandlingPath isn't the only option, remove it from the message.

Reply:
How else is it configured? It looks like you either configure the path or you explicitly provide a handler. In the latter case, I don't see how it could 404 except for what I mentioned in https://github.com/dotnet/aspnetcore/pull/25062#discussion_r473497042

Reply:
That handler can be a full pipeline which would still end in a default 404.